server/src/internalClusterTest/java/org/opensearch/remotemigration/DocrepToRemoteDualReplicationIT.java

```diff
@@ -0,0 +1,420 @@
+                RetentionLeases retentionLeases = shardStats.getRetentionLeaseStats().retentionLeases();
+                if (shardRouting.primary()) {
+                    // Primary copy should be on remote node and should have retention leases
+                    assertTrue(discoveryNode.isRemoteStoreNode());
+                    assertRetentionLeaseConsistency(shardStats, retentionLeases);
+                } else if (discoveryNode.isRemoteStoreNode()) {
+                    // Replica copy on remote node should not have retention leases
+                    assertTrue(shardStats.getRetentionLeaseStats().retentionLeases().leases().isEmpty());
+                } else {
+                    // Replica copy on docrep node should have retention leases
+                    assertRetentionLeaseConsistency(shardStats, retentionLeases);
+                }
+            }
+        }, 40, TimeUnit.SECONDS);
+    }
+
+    /*
+    Scenario:
+    - Starts 1 docrep backed data node
+    - Creates an index with 0 replica
+    - Starts 1 remote backed data node
+    - Move primary copy from docrep to remote through _cluster/reroute
+    - Expands index to 1 replica
+    - Stops remote enabled node
+    - Ensure doc count is same after failover
+    - Index some more docs to ensure working of failed-over primary
+     */
+    public void testFailoverRemotePrimaryToDocrepReplica() throws Exception {
+        internalCluster().setBootstrapClusterManagerNodeIndex(0);
+        internalCluster().startClusterManagerOnlyNode();
+
+        logger.info("---> Starting 1 docrep data nodes");
+        String docrepNodeName = internalCluster().startDataOnlyNode();
+        internalCluster().validateClusterFormed();
+        assertEquals(internalCluster().client().admin().cluster().prepareGetRepositories().get().repositories().size(), 0);
+
+        logger.info("---> Creating index with 0 replica");
+        Settings excludeRemoteNode = Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0).build();
+        createIndex(FAILOVER_REMOTE_TO_DOCREP, excludeRemoteNode);
+        ensureGreen(FAILOVER_REMOTE_TO_DOCREP);
+        initDocRepToRemoteMigration();
+        logger.info("---> Starting 1 remote enabled data node");
+        addRemote = true;
+        String remoteNodeName = internalCluster().startDataOnlyNode();
+        internalCluster().validateClusterFormed();
+        assertEquals(
+            internalCluster().client()
+                .admin()
+                .cluster()
+                .prepareGetRepositories(REPOSITORY_NAME, REPOSITORY_2_NAME)
+                .get()
+                .repositories()
+                .size(),
+            2
+        );
+
+        int firstBatch = randomIntBetween(1, 100);
+        logger.info("---> Indexing {} docs", firstBatch);
+        SyncIndexingService indexingService = new SyncIndexingService(FAILOVER_REMOTE_TO_DOCREP, firstBatch);
+        indexingService.startIndexing();
```

Comment:
Could we try background indexing and verify that the total successfully indexed doc count is same as the total docs in the index?